debug-and-error-logs.md

```diff
@@ -0,0 +1,115 @@
+
+**Default Behavior by Severity Level**
+
+Command | `INFO` messages | `WARNING` and above messages
+--------|--------|--------------------
+[`cockroach start`](start-a-node.html) | Logged to file | Logged to file &amp; `stderr`
+[All other commands](cockroach-commands.html) | Discarded | Logged to `stderr`
+
+### Output Locations
+
+Based on the command's flags and the message's severity level, CockroachDB does one of the following: 
+
+- [Writes the message to a file](#write-to-file)
+- [Prints it to `stderr`](#print-to-stderr)
+- [Discards it entirely](#discard-message)
+
+#### Write to File
+
+When file logging is enabled (by default for `cockroach start` or by including the `--log-dir` [flag](#flags) for other commands), CockroachDB writes messages to files in the directory specified by `--log-dir`. 
+
+The log's filenames use the following format:
+
+~~~
+cockroach.[host].[user].[start date/time + process ID].log
+~~~
+
+You can control the severity level of which messages are written to files with the `--log-file-verbosity` [flag](#flags).
+
+**Defaults per Command**
+
+Command | Log file location (`--log-dir`)
+--------|-------------|
+[`cockroach start`](start-a-node.html) | `--log-dir=<first `[`store`](start-a-node.html#store)` dir>/logs`<br/><br/>This location is also printed in the **Logs** field of the [standard output](start-a-node.html#standard-output) of `cockroach start`.
+[All other commands](cockroach-commands.html) | *N/A (Messages are not written to files. To log messages from commands besides `start`, set the `--log-dir` [flag](#flags) to a directory.)* |
+
+#### Print to `stderr`
+
+`stderr` is a standard Unix interface, which normally prints messages to a machine's console.
+
+By default, all commands print messages with a [severity level](#severity-levels) `WARNING` or greater to `stderr`. However, you can control the severity level of messages printed to `stderr` with the `--logtostderr` [flag](#flags).
+
+#### Discard Message
+
+Messages with severity levels below the `--logtostderr` and `--log-file-verbosity` values are neither written to files nor printed to `stderr`, so they are discarded.
+
+By default, commands besides `cockroach start` have their messages with the `INFO` [severity level](#severity-levels) totally discarded.
+
+## Flags
+
+{% include custom/logging-flags.md %}
+
+The `--log-backtrace-at` and `--verbosity` flags are intended for internal debugging by CockroachDB contributors.
+
+## Log Queries
+
+To help troubleshoot [query performance issues](query-behavior-troubleshooting.html#performance-issues), you can enable cluster-wide error logging for long-running SQL queries or all queries, regardless of time.
+
+- **Long-running queries**:
+  
+  ~~~ sql
+  > SET CLUSTER SETTING sql.trace.txn.threshold = '[time]';
+  ~~~
+
+  The `[time]` parameter accepts common time specifiers, such as `100ms` or `2s`.
+
+  Disable this logging by setting `[time]` to `0s`.
```

Comment:
Mention here that setting this will make *all* queries quite slower and the nodes will consume much more memory overall.